StyleCop.Analyzers/StyleCop.Analyzers.Test/OrderingRules/SA1200OutsideNamespaceUnitTests.cs

```diff
@@ -0,0 +1,214 @@
+{
+}";
+
+        private const string InterfaceDefinition = @"public interface TestInterface
+{
+}";
+
+        private const string EnumDefinition = @"public enum TestEnum
+{
+    TestValue
+}";
+
+        private const string DelegateDefinition = @"public delegate void TestDelegate();";
+
+        /// <summary>
+        /// Verifies that using statements in a namespace produces the expected diagnostics.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestInvalidUsingStatementsInNamespaceAsync()
+        {
+            var testCode = @"namespace TestNamespace
+{
+    using System;
+    using System.Threading;
+}
+";
+            var fixedTestCode = @"using System;
+using System.Threading;
+
+namespace TestNamespace
+{
+}
+";
+
+            DiagnosticResult[] expectedResults =
+            {
+                this.CSharpDiagnostic(SA1200UsingDirectivesMustBePlacedCorrectly.DescriptorOutside).WithLocation(3, 5),
+                this.CSharpDiagnostic(SA1200UsingDirectivesMustBePlacedCorrectly.DescriptorOutside).WithLocation(4, 5)
+            };
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expectedResults, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpDiagnosticAsync(fixedTestCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpFixAsync(testCode, fixedTestCode).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that simplified using statements in a namespace are expanded during the code fix operation.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestInvalidSimplifiedUsingStatementsInNamespaceAsync()
+        {
+            var testCode = @"namespace System
+{
+    using System;
+    using System.Threading;
+    using Reflection;
+}
+";
+            var fixedTestCode = @"using System;
+using System.Reflection;
+using System.Threading;
+
+namespace System
+{
+}
+";
+
+            DiagnosticResult[] expectedResults =
+            {
+                this.CSharpDiagnostic(SA1200UsingDirectivesMustBePlacedCorrectly.DescriptorOutside).WithLocation(3, 5),
+                this.CSharpDiagnostic(SA1200UsingDirectivesMustBePlacedCorrectly.DescriptorOutside).WithLocation(4, 5),
+                this.CSharpDiagnostic(SA1200UsingDirectivesMustBePlacedCorrectly.DescriptorOutside).WithLocation(5, 5),
+            };
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expectedResults, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpDiagnosticAsync(fixedTestCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpFixAsync(testCode, fixedTestCode).ConfigureAwait(false);
+        }
```

Comment:
Could you also write a test that use the following test code:
```C#
namespace System.MyExtension
{
    using System.Threading;
    using Reflection;
}
```

It should be fixed as:
```C#
using System.Reflection;
using System.Threading;

namespace System.MyExtension
{
}
```